validate-differential-expression/gene_DE_validate.Rmd

```diff
@@ -0,0 +1,352 @@
+---
+title: "Comparing Differential Expression Analysis"
+author: "ALSF CCDL - Candace Savonen"
+output:   
+  html_notebook: 
+    toc: true
+    toc_float: true
+---
+
+*Purpose*: This notebook takes data processed by refine.bio and compares the 
+differential expression results to the author-submitted versions of the data.
+
+## 1) Set up
+
+```{r}
+if (!("limma" %in% installed.packages())) {
+  # Install limma
+  BiocManager::install("limma", update = FALSE)
+}
+if (!("VennDiagram" %in% installed.packages())) {
+  # Install VennDiagram
+  BiocManager::install("VennDiagram", update = FALSE)
+}
+```
+
+Attach the `limma` library:
+
+```{r}
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Attach library
+library(limma)
+```
+
+Set up output directories.
+
+```{r}
+# Make a results directory if it isn't created yet
+if (!dir.exists("results")) {
+  dir.create("results")
+}
+# Make a plots directory if it isn't created yet
+if (!dir.exists("plots")) {
+  dir.create("plots")
+}
+```
+
+## 2) Import our separate differential expression results
+Import the `limma` results output we created in `author_processed_DE.Rmd`.
+
+```{r}
+# Load in the summary statistics from limma run
+author.results <- readr::read_tsv(file.path("results", "GSE37382_limma_results.tsv")) 
+```
+
+## 3) Convert these Affymetrix ids into Ensembl gene IDs (what refine.bio datasets have)
+Here we need to convert our Affymetrix transcript cluster ids into Ensembl gene
+ids. 
+If you are using this notebook for your own data, and need more information on 
+`ensembl_id_convert.Rmd` in this repository. 
+You can also use the [id conversion repository prepared by the CCDL.](https://github.com/AlexsLemonade/identifier-refinery) 
+The pre-made id conversion key files themselves for refine.bio supported 
+microarray platforms can be downloaded from [Zenodo](https://zenodo.org/record/1322711#.XLdJb-tKiXE). 
+
+```{r}
+if (file.exists("hugene11st.tsv.zip")) {
+  # Unzip hugene11st id conversion file obtained from identifier refinery
+  unzip(file.path("data", "hugene11st.tsv.zip"))
+}
+
+# Read in the probe conversion info - this step may take some time
+probe.conversion <- readr::read_tsv(file.path("data", "hugene11st.tsv"), 
+                                    progress = FALSE) %>% 
+  dplyr::select("PROBEID", "ENSEMBL", "SYMBOL")
+
+# Add annotation to imported author limma results 
+author.full.results <- author.results %>%
+  dplyr::inner_join(probe.conversion, by = c("affy_probe_ids" = "PROBEID")) %>%
+  dplyr::mutate("ENSEMBL" = as.factor(ENSEMBL))
+```
+
+Summarize probe sets that all correspond to the same gene. 
```

Comment:
Yes. There are 1:many mappings for most probes to gene IDs. It all depends on how we want to summarize the info. There's a lot of ways we could do it. I can do the way you mentioned here but how do I summarize the p values or what do I do with 'ties' (where half the probes go one way and the other half go the other direction).

Reply:
If we really want to be realistic about it, we should summarize the probe data before DE or use the all the probe data in the DE tests (like MANOVA or something).

Reply:
When I say list, I mean "set of genes or probes" that are differentially expressed where I'm using some cutoff and ignoring the p-values themselves. So in the example above, I'm proposing that you would convert your 'up' list of probes to Ensembl gene IDs and convert your 'down' list to Ensembl gene IDs. If there are Ensembl gene IDs that are in both your up and down lists, discard them because there's disagreement.

Reply:
Hm. Okay I could try that.

Reply:
Per our in person conversation, I've changed the up and down gene lists to be made only when the probe sets have no conflicting directions. So up-regulated genes do not have any probesets that are down-regulated and vice versa.